### PR TITLE
Fix crash in iOS accessibility

### PR DESCRIPTION
**Pull request by goderbauer**
Previously, whenever an action would remove the focused element from the screen the app would crash. This change tells iOS to focus the first on-screen element after every change to the semantics tree. This avoids the crash.

In a future iteration, we should tell iOS which element it needs to focus by looking at the semantics tree (instead of leaving it up to the iOS).

### CODE REVIEW

File: shell/platform/darwin/ios/framework/Source/accessibility_bridge.mm [hunk 1]
```diff
@@ -257,7 +257,10 @@ - (BOOL)accessibilityPerformMagicTap {
   } else {
     view_.accessibilityElements = nil;
   }
-  UIAccessibilityPostNotification(UIAccessibilityLayoutChangedNotification, nil);
+  // TODO(goderbauer): If the previously focused element is still on screen we should fire a
+  //     UIAccessibilityLayoutChangedNotification instead. If the element is no longer on screen
+  //     we should use the semantics tree to figure out which element to focus instead.
+  UIAccessibilityPostNotification(UIAccessibilityScreenChangedNotification, nil);
```
Comment (Hixie) [inline shell/platform/darwin/ios/framework/Source/accessibility_bridge.mm]: We can tell if the focused element is on the screen (i.e. has non-zero dimensions and is still in the semantics tree). Would it be worth checking for this in this patch?

Comment (goderbauer) [inline shell/platform/darwin/ios/framework/Source/accessibility_bridge.mm]: Added. PTAL.

**Comment by Hixie** [comment]
This seems good to me, but I'm not a good person to review ObjectiveC code. RSLGTM, cc @chinmay or @cbracken for better review.

File: shell/platform/darwin/ios/framework/Source/accessibility_bridge.mm [hunk 2]
```diff
@@ -247,17 +255,26 @@ - (BOOL)accessibilityPerformMagicTap {
...
+    //     UIAccessibilityLayoutChangedNotification with that element instead.
+    UIAccessibilityPostNotification(UIAccessibilityScreenChangedNotification, nil);
   } else {
-    view_.accessibilityElements = nil;
+    UIAccessibilityPostNotification(UIAccessibilityLayoutChangedNotification, nil);
   }
-  UIAccessibilityPostNotification(UIAccessibilityLayoutChangedNotification, nil);
+
+  ReleaseObjects(doomed_objects);
+  doomed_objects.clear();
```
Comment (cbracken) [inline shell/platform/darwin/ios/framework/Source/accessibility_bridge.mm]: Maybe do this in `ReleaseObjects` and kill this line here and in the dtor? I can't think of a situation where you'd want a set of freed object pointers.

Comment (goderbauer) [inline shell/platform/darwin/ios/framework/Source/accessibility_bridge.mm]: Done.

File: shell/platform/darwin/ios/framework/Source/accessibility_bridge.mm [hunk 3]
```diff
@@ -247,17 +255,26 @@ - (BOOL)accessibilityPerformMagicTap {
...
+  }
+
+  if (doomed_focused_object != nil) {
+    // Previously focused element is no longer in the tree, let iOS figure out what to focus next.
+    // TODO(goderbauer): Figure out which element should be focused next and post
+    //     UIAccessibilityLayoutChangedNotification with that element instead.
+    UIAccessibilityPostNotification(UIAccessibilityScreenChangedNotification, nil);
   } else {
-    view_.accessibilityElements = nil;
+    UIAccessibilityPostNotification(UIAccessibilityLayoutChangedNotification, nil);
```
Comment (goderbauer) [inline shell/platform/darwin/ios/framework/Source/accessibility_bridge.mm]: Clarified what `nil` means in the comments.

File: shell/platform/darwin/ios/framework/Source/accessibility_bridge.mm [hunk 4]
```diff
@@ -247,17 +255,27 @@ - (BOOL)accessibilityPerformMagicTap {
...
+    SemanticsObject* object = entry.second;
+    if ([object accessibilityElementIsFocused]) {
+      doomed_focused_object = object;
+      break;
     }
+  }
+
+  if (doomed_focused_object != nil) {
+    // Previously focused element is no longer in the tree.
+    // Passing `nil` as argument to let iOS figure out what to focus next.
```
Comment (cbracken) [inline shell/platform/darwin/ios/framework/Source/accessibility_bridge.mm]: Interesting - where did you find this?

Comment (goderbauer) [inline shell/platform/darwin/ios/framework/Source/accessibility_bridge.mm]: From the Apple docs it wasn't clear to my what the difference between `UIAccessibilityScreenChangedNotification` and `UIAccessibilityLayoutChangedNotification` was supposed to be. After some searching, I found [this post](https://stackoverflow.com/questions/27797515/actual-difference-between-uiaccessibilitylayoutchangednotification-and-uiaccessi) which describes it pretty good.

**Pull request merged**